lib/overcommit/hook_context/pre_commit.rb

```diff
@@ -7,7 +7,21 @@ module Overcommit::HookContext
   # This includes staged files, which lines of those files have been modified,
   # etc. It is also responsible for saving/restoring the state of the repo so
   # hooks only inspect staged changes.
-  class PreCommit < Base
+  class PreCommit < Base # rubocop:disable ClassLength
+    # Returns whether this hook run was triggered by `git commit --amend`
+    def amend?
```

Comment:
The tense of this verb doesn't seem right given the context we're in. We're trying to return whether this `pre-commit` was the result of an amendment, so it feels like we should call it `amendment?` (making it act like a "is a" predicate). This also reads slightly better when used in `if` statements in hooks.

Naming things is frustrating. :)